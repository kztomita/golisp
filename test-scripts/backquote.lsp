(setq a 1)
(setq b 2)
(setq c 3)
(setq d '(1 2 3))
(print `(a b))
(print `a)
(print `(a ,b c))
(print `(a (,b c)))
(print `(a (quote ,b)))
(print `(a ',b))
(print `(a b ,c (',(+ a b c)) (+ a b) 'c '((,a ,b))))
(print `(a ,b `(,b ,,c)))
(print `(a ,d ,@d))
(print `(a ,b `(,@d ,,@d)))

