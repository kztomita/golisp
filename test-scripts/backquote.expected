
(a b) 
a 
(a 2 c) 
(a (2 c)) 
(a (quote 2)) 
(a (quote 2)) 
(a b 3 ((quote 6)) (+ a b) (quote c) (quote ((1 2)))) 
(a 2 (list b 3)) 
(a (1 2 3) 1 2 3) 
(a 2 (append d (list 1 2 3))) 
